src/Chainweb/Miner/Coordinator.hs

```diff
@@ -144,10 +154,10 @@ newWork logFun mode choice miner@(Miner mid _) pact tpw c = do
     --
     let !p = ParentHeader (c ^?! ixg cid)
 
-    bool (public p) (private cid p <$> readTVarIO tpw) (mode == Private) >>= \case
+    either (public p) (\m -> primed m cid p <$> readTVarIO tpw) eminer >>= \case
         -- The proposed Chain wasn't mineable, either because the adjacent
         -- parents weren't available, or because the chain is mid-update.
-        Nothing -> newWork logFun mode (TriedLast cid) miner pact tpw c
+        Nothing -> newWork logFun mode (TriedLast cid) eminer pact tpw c
```

Comment:
Unrelated to this PR: I wonder what happens if all but one chain is blocked. Would this code would spin while pact computes newBlock. That might cause performance issues. I think that this is rare. But I wonder if we should detect and log that condition.

In the primed case, I wonder if the next chain could be randomly chosen from the available chains in primed work. If no work is available in `PrimedWork` the code could await a change to the PrimedWork variable, thus avoiding to spin.

Reply:
> I wonder what happens if all but one chain is blocked. 

Can you describe when such a thing would happen?

Reply:
I don't know how often this happens, but just think of a process that mines on all chains but one. At some point all chains would be blocked waiting for the one chain. I hope that this is very rare with random chain selection, but it has a non-zero probability.